core/services/fluxmonitorv2/integrations_test.go

```diff
@@ -0,0 +1,946 @@
+		generatePriceResponseFn(&reportPrice),
+	)
+	t.Cleanup(mockServer.Close)
+
+	// Generate custom TOML for this test due to precision change
+	toml := `
+type              = "fluxmonitor"
+schemaVersion     = 1
+name              = "example flux monitor spec"
+contractAddress   = "%s"
+precision = %d
+threshold = 0.5
+absoluteThreshold = 0.01
+
+idleTimerPeriod = "1h"
+idleTimerDisabled = false
+
+pollTimerPeriod = "%s"
+pollTimerDisabled = false
+
+observationSource = """
+ds1 [type=http method=GET url="%s"];
+ds1_parse [type=jsonparse path="data,result"];
+
+ds1 -> ds1_parse
+"""
+`
+
+	s := fmt.Sprintf(toml, fa.aggregatorContractAddress, 8, "100ms", mockServer.URL)
+
+	// raise flags
+	fa.flagsContract.RaiseFlag(fa.sergey, utils.ZeroAddress) // global kill switch
+	fa.flagsContract.RaiseFlag(fa.sergey, fa.aggregatorContractAddress)
+	fa.backend.Commit()
+
+	requestBody, err := json.Marshal(models.CreateJobSpecRequest{
+		TOML: string(s),
+	})
+	assert.NoError(t, err)
+
+	j := cltest.CreateJobViaWeb2(t, app, string(requestBody))
+
+	go func() {
+		for {
+			fa.backend.Commit()
+			time.Sleep(500 * time.Millisecond)
+		}
+	}()
+
+	// We should see a spec error because the value is too large to submit on-chain.
+	jobID, err := strconv.ParseInt(j.ID, 10, 32)
+	require.NoError(t, err)
+
+	jse := cltest.WaitForSpecErrorV2(t, app.Store, int32(jobID), 1)
+	assert.Contains(t, jse[0].Description, "Answer is outside acceptable range")
+}
+
+func TestFluxMonitorAntiSpamLogic(t *testing.T) {
+	// - deploy a brand new FM contract
+	fa := setupFluxAggregatorUniverse(t)
+
+	// - add oracles
+	oracleList := []common.Address{fa.neil.From, fa.ned.From, fa.nallory.From}
+	_, err := fa.aggregatorContract.ChangeOracles(fa.sergey, emptyList, oracleList, oracleList, 2, 3, 2)
+	assert.NoError(t, err, "failed to add oracles to aggregator")
+	fa.backend.Commit()
+	checkOraclesAdded(t, fa, oracleList)
+
+	// Set up chainlink app
+	app := setupApplication(t, fa, func(cfg *orm.Config) {
+		cfg.Set("DEFAULT_HTTP_TIMEOUT", "100ms")
+		cfg.Set("TRIGGER_FALLBACK_DB_POLL_INTERVAL", "1s")
+	})
+	require.NoError(t, app.StartAndConnect())
+
+	minFee := app.Store.Config.MinimumContractPayment().ToInt().Int64()
+	require.Equal(t, fee, minFee, "fee paid by FluxAggregator (%d) must at "+
+		"least match MinimumContractPayment (%s). (Which is currently set in "+
+		"cltest.go.)", fee, minFee)
+
+	answer := int64(1) // Answer the nodes give on the first round
+
+	//- have one of the fake nodes start a round.
+	roundId := int64(1)
+	processedAnswer := answer * 100 /* job has multiply times 100 */
+	submitAnswer(t, answerParams{
+		fa:              &fa,
+		roundId:         roundId,
+		answer:          processedAnswer,
+		from:            fa.neil,
+		isNewRound:      true,
+		completesAnswer: false,
+	})
+
+	// - successfully close the round through the submissions of the other nodes
+	// Response by malicious chainlink node, nallory
+	initialBalance := currentBalance(t, &fa).Int64()
```

Comment:
could use a comment about where this initialBalance comes from / represents - took us a little bit of time to figure out

Reply:
Added a comment